packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -676,14 +682,18 @@ function normalizeToEnvName(input: string): string {
   return input.toUpperCase().replace(/-/g, "_").replace(/\./g, "__"); // replace "-" to "_" and "." to "__"
 }
 
-function generateResourceName(ctx: SolutionContext): void {
+function generateResourceBaseName(appName: string, envName: string): string {
   const maxAppNameLength = 10;
-  const appName = ctx.projectSettings!.appName;
-  const suffix = ctx.envInfo.profile.get(GLOBAL_CONFIG)?.getString("resourceNameSuffix");
   const normalizedAppName = appName.replace(/[^a-zA-Z0-9]/g, "").toLowerCase();
+  return normalizedAppName.substr(0, maxAppNameLength) + envName.substr(0, maxAppNameLength);
+}
+
+function generateResourceName(ctx: SolutionContext): void {
+  const appName = ctx.projectSettings!.appName;
+  const suffix = ctx.envInfo!.envName;
   ctx.envInfo.profile
     .get(GLOBAL_CONFIG)
-    ?.set("resource_base_name", normalizedAppName.substr(0, maxAppNameLength) + suffix);
+    ?.set("resource_base_name", generateResourceBaseName(appName, suffix));
```

Comment:
since the resource base name is now an actual value instead of placeholder in parameter file, maybe we do not need to add this config to profile anymore?

Reply:
sure